dashboard/test/ui/test_runner/test/test_runner_options_parser_test.rb

```diff
@@ -1,8 +1,64 @@
 require 'minitest/autorun'
 require_relative '../test_runner_options_parser'
 class TestRunnerOptionsParserTest < Minitest::Test
+  BROWSERS_JSON_CONTENT = [
```

Comment:
I'm not entirely sure it's valuable to test this so exhaustively; it feels pretty silly (and potentially over-fragile) to duplicate all of our existing config here in the test like this.

I think there is opportunity here for us to write a small number of tests that check for individual, specific things we know we care about in this config. I'd definitely like to see a test like:

```ruby
  def test_mobile_vs_desktop_safari
    # Selenium uses capitalization to distinguish between mobile and desktop
    # Safari; make sure our configuration reflects that
    parser = TestRunnerOptionsParser.new(%w[-c Safari,iPhone,iPad])
    parser.parse
    assert_equal %w[Safari safari safari], desktop_parser.select_browser_configs.pluck(:browserName)
  end
```

Reply:
I love your approach!

I think the current test (with the long expected object)
1) Forces developers to validate the value is parsed for that specific config (Issue, this test wouldn't cover new cases, although this code doesnt change often)
2) When I wrote them my logic was "Well if someone changes, they gotta validate the code process the field properly. This is from experience where I have seen small config changes to a json or xml and somehow they break the parsing of the specific config.

Let me put some thoughts on point #1

After reading these thoughts, let me know if #2 makes sense, if not, I am not strongly married to it